### PR TITLE
wss switch on client

### PR DESCRIPTION
**Pull request by radutopala**

### CODE REVIEW

**Comment by alexellis** [comment]
Radu please read the contribution guide.

I'll close the PR feel free to fix and edit the description etc. 

Alex

**Comment by alexellis** [comment]
Ping @radutopala

File: pkg/client/client.go [hunk 1]
```diff
@@ -36,7 +36,11 @@ func (c *Client) Connect() error {
 		return http.ErrUseLastResponse
 	}
 
-	u := url.URL{Scheme: "ws", Host: c.Remote, Path: "/tunnel"}
+	remote, _ := url.Parse(c.Remote)
+	if !strings.HasPrefix(remote.Scheme, "ws") {
```
Comment (alexellis) [inline pkg/client/client.go]: When testing this it failed with:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1238b7a]
```

Comment (alexellis) [reply pkg/client/client.go]: Fixed via: https://github.com/alexellis/inlets/commit/d745f01850b54aa4daa3a701c41cc69e3282af91

Please do contribute again and see https://github.com/alexellis/inlets/blob/master/CONTRIBUTING.md for more details about that.

Comment (radutopala) [reply pkg/client/client.go]: @alexellis 👍 .. sorry, didn't have enough time to push a ticket and describe better. But you've got the idea 😄 
Would be awesome if you'd add a proxy for websockets as well, beyond http => #13

**Comment by alexellis** [comment]
How would you see that working with the current architecture?

**Comment by radutopala** [comment]
You would call ./inlets with -remote=wss://host

Or were you asking about proxying websockets (#13)? If yes, then I guess
based on the url.Scheme, we could launch a different dialer for ws, on the
same upstream host.

- Radu Topala, Trisoft

**Pull request closed without merging**